java-plugin/src/main/java/fr/greencodeinitiative/java/checks/ForceUsingLazyFetchTypeInJPAEntity.java

```diff
@@ -0,0 +1,97 @@
+package fr.greencodeinitiative.java.checks;
+
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.*;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+
+import java.util.List;
+
+@Rule(key = "EC80", name = "Developpement",
+        description = ForceUsingLazyFetchTypeInJPAEntity.MESSAGERULE,
+        priority = Priority.MINOR,
+        tags = {"bug"})
+public class ForceUsingLazyFetchTypeInJPAEntity extends IssuableSubscriptionVisitor {
+
+    protected static final String MESSAGERULE = "Force the use of LAZY FetchType";
+    private static final String EAGER_KEYWORD = "EAGER";
+    private static final String FETCH_KEYWORD = "fetch";
+    private static final String ONE_TO_MANY = "OneToMany";
+    private static final String MANY_TO_ONE = "ManyToOne";
+    private static final String ONE_TO_ONE = "OneToOne";
+    private static final String MANY_TO_MANY = "ManyToMany";
+
+    @Override
+    public List<Kind> nodesToVisit() {
+        return List.of(Kind.VARIABLE);
+    }
+
+    @Override
+    public void visitNode(Tree tree) {
+        VariableTree variableTree = (VariableTree) tree;
+        ModifiersTree modifierTrees = variableTree.modifiers();
+        if (modifierTrees == null) {
+            return;
+        }
+        List<AnnotationTree> annotations = modifierTrees.annotations();
+        // get all annotations on the attribute
+        for (AnnotationTree annotationTree : annotations) {
+            if (!needToCheckExplicitAnnotation(annotationTree.annotationType().symbolType().name())) {
+                // no Explicit annotation (@OneToMany, @ManyToOne, @ManyToMany, @OneToOne) was found,
+                // and we don't need to investigate further
+                continue;
+            }
+            Arguments arguments = annotationTree.arguments();
+            performsCheck(arguments, annotationTree, tree);
+        }
+    }
+
+    /**
+     * perform the check for two case :
+     * fetch keyword is found -> parse the annotation argument and search for eager keyword
+     * fetch keyword was not found -> we use the default value of fetch type by join type :
+     * OneToMany: Lazy
+     * ManyToOne: Eager
+     * ManyToMany: Lazy
+     * OneToOne: Eager
+     * reporting the issues if necessary
+     */
+    private void performsCheck(Arguments arguments, AnnotationTree annotationTree, Tree tree) {
```

Comment:
here, no need to give `arguments` as input parameter because it already is present inside  `annotationTree` input parameter